Modules/rat_rescue.py

```diff
@@ -146,6 +152,29 @@ def __eq__(self, other) -> bool:
 
             return all(conditions)
 
+    def __hash__(self):
+
+        attributes = (
+            self.case_id,
+            self.board_index,
+            self.client,
+            self.platform,
+            self.first_limpet,
+            self.created_at,
+            self.updated_at,
+            self.system,
+            self.active,
+            self.code_red,
+            self.outcome,
+            self.title,
+            self.first_limpet,
+            self.lang_id,
+            self.irc_nickname,
+        )
+        if self._hash is None:
+            self._hash = reduce(xor, map(hash, attributes))
```

Comment:
By moving the tuple creation into the `if` block, you could save it from being executed when the hash is already available.